cmd/homi/setup/cmd.go

```diff
@@ -546,11 +550,24 @@ func gen(ctx *cli.Context) error {
 	baobabTest := ctx.Bool(baobabTestFlag.Name)
 	cypress := ctx.Bool(cypressFlag.Name)
 	cypressTest := ctx.Bool(cypressTestFlag.Name)
+	clique := ctx.Bool(cliqueFlag.Name)
 	serviceChain := ctx.Bool(serviceChainFlag.Name)
 	serviceChainTest := ctx.Bool(serviceChainTestFlag.Name)
 	chainid := ctx.Uint64(chainIDFlag.Name)
 	serviceChainId := ctx.Uint64(serviceChainIDFlag.Name)
 
+	// Note-klaytn : the following code that seems unnecessary is for the priority to flags, not yaml
+	if !baobab && !baobabTest && !cypress && !cypressTest && !serviceChain && !serviceChainTest && !clique {
+		genesisType := ctx.String(genesisTypeFlag.Name)
+		baobab = genesisType == "baobab"
+		baobabTest = genesisType == "baobab-test"
+		cypress = genesisType == "cypress"
+		cypressTest = genesisType == "cypress-test"
+		serviceChain = genesisType == "servicechain"
+		serviceChainTest = genesisType == "servicechain-test"
+		clique = genesisType == "clique"
```

Comment:
```suggestion
	switch genesisType := ctx.String(genesisTypeFlag.Name); genesisType {
	case "baobab":
		baobab = true
	case "baobab-test":
		baobabTest = true
	case "cypress":
		cypress = true	
	case "cypress-test":
		cypressTest = true	
	case "servicechain":
		servicechain = true	
	case "servicechain-test":
		serviceChainTest = true		
	case "clique":
		clique = true		
	default:
		fmt.Printf("Unknown genesis type is %s.\n", genesisType)
	}
```
I like this switch-case style a bit more. You can decide at your convenience.

Reply:
Looks better thanks!